roles/insert_dns_records/tasks/configure_firewall.yml

```diff
@@ -20,3 +20,39 @@
     - internal
     - public
   when: use_dhcp == true
+
+- name: Open port in firewall for PXE
+  ansible.posix.firewalld:
+    port: "4011/udp"
+    permanent: yes
+    immediate: yes
+    state: enabled
+    zone: "{{ item }}"
+  loop:
+    - internal
+    - public
+  when: use_pxe == true
+
+- name: Open port in firewall for PXE
+  ansible.posix.firewalld:
+    port: "81/tcp"
+    permanent: yes
+    immediate: yes
+    state: enabled
+    zone: "{{ item }}"
+  loop:
+    - internal
+    - public
+  when: use_pxe == true
+
+- name: Open port in firewall for PXE
+  ansible.posix.firewalld:
+    port: "69/udp"
+    permanent: yes
+    immediate: yes
+    state: enabled
+    zone: "{{ item }}"
+  loop:
+    - internal
+    - public
+  when: use_pxe == true
```

Comment:
It should be possible to merge all these tasks into one, iterating over a list of ports.
```yaml
pxe_required_open_ports:
  - "4011/udp"
  - "81/tcp"
  - "69/udp"
```

You'll probably need to use a nested loop for that: https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#iterating-over-nested-lists

Reply:
81 I will delete because we don’t use that we are using the http_store